public/app/plugins/datasource/influxdb/components/InfluxLogsQueryField.tsx

```diff
@@ -60,6 +60,10 @@ export class InfluxLogsQueryField extends React.PureComponent<Props, State> {
     });
   };
 
+  private pairsSet = (pairs: KeyValuePair[]) => {
+    return pairs.length && pairs.reduce((prev, pair) => prev && !!pair.key && !!pair.operator && !!pair.value, true);
```

Comment:
I'd probably use `Array​.prototype​.every()` here instead of `reduce`. Then you might be able to get rid of the `!!` as well.